### ISSUE TITLE
Question: Is it possible to encrypt an existing non encrypted database?

### ISSUE THREAD

**Issue by Devilcow** [praeclarum/sqlite-net#1108]
I have a released app that I would like to add encryption to the database in a future release. Is it possible to take an existing db and add encryption to it? 
Seeing as I am a disabled vet, $2,000 for a license to get the official SQLite SEE extension is about $1960 more than I can afford lol. 

It would be cool if there were a way to overlay the encryption on the existing database. My code creates the database from scratch when it runs for the first time (also makes schema updates easy since I can just add the schema changes to my code and have it run on the first load for that version). Anyhow, I could create a new database with the encryption enabled, and then copy the records across, but it would be nice to not have to do so.

I am working on starting another program in WPF, but have created "base" db3 files that already have encryption on them, so all's good there (I hope, so far I've created the db files, and project files only, so I have plenty of opportunity for that to become another disastrous nightmare HAHA!)

Cheers!

**Comment by praeclarum** [comment]
You can use the sqlite-net-sqlcipher package to get encryption. I apologize, but I can't remember the exact right way to migrate. The cipher docs say you have to manually copy the database objects: https://discuss.zetetic.net/t/how-to-encrypt-a-plaintext-sqlite-database-to-use-sqlcipher-and-avoid-file-is-encrypted-or-is-not-a-database-errors/868

It seems there is a new `Export` function that I can add to this library to make it easier, but, for now, a manual copy of all the objects seems necessary. :-(

**Comment by sjlombardo** [comment]
There is also a convenience SQL function exposed by SQLCipher called `sqlcipher_export`. It can be used to completely reproduce a plaintext database into an attached encrypted database for migration purposes:

https://www.zetetic.net/sqlcipher/sqlcipher-api/#sqlcipher_export

**Comment by praeclarum** [comment]
@sjlombardo Yeah I added a bug to implement that: https://github.com/praeclarum/sqlite-net/issues/1109

**Comment by Devilcow** [comment]
Excellent! Thanks!
